io/EptReader.cpp

```diff
@@ -767,5 +769,82 @@ Dimension::Type EptReader::getCoercedTypeTest(const NL::json& j)
     return getCoercedType(j);
 }
 
+//============================================================================================
+// Pravin: APIs from here are written to make EPT Reader streamable.
+
+void EptReader::loadNextOverlap()
+{
+    std::map<Key, uint64_t>::iterator itr = m_overlaps.begin();
+    std::advance(itr, m_nodeId-1);
+    Key key(itr->first);
+    log()->get(LogLevel::Debug)
+        << "Streaming Data " << m_nodeId << "/" << m_overlaps.size() << ": "
+        << key.toString() << std::endl;
+
+    uint64_t startId(0);
+	m_pointTable.reset(new PointTable());
+	PointLayoutPtr layout = m_pointTable->layout();
+    for (auto id : m_remoteLayout->dims())
+    {
+        layout->registerOrAssignDim(m_remoteLayout->dimName(id),
+                                    m_remoteLayout->dimType(id));
+	}
+
+	m_pointView.reset(new PointView(*m_pointTable));
+
+	if (m_info->dataType() == EptInfo::DataType::Laszip)
+    {
+		startId=readLaszip(*m_pointView, key, m_nodeId);
+    }
+    else
+    {
+        startId = readBinary(*m_pointView, key, m_nodeId);
+	}
+    m_currentIndex = 0;
+
+    // Read addon information after the native data, we'll possibly
+    // overwrite attributes.
+    for (const auto& addon : m_addons)
+    {
+        readAddon(*m_pointView, key, *addon, startId);
+    }
+    m_nodeId++;
+}
+
+void EptReader::fillPoint(PointRef& point)
+{
+    DimTypeList dims = m_pointView->dimTypes();
+	char* buffer = new char[m_pointView->pointSize()];
+    m_pointView->getPackedPoint(dims, m_currentIndex++, buffer);
```

Comment:
take the m_m_currentIndex++ out of the function call please

Reply:
Done!